### ISSUE TITLE
False positive from `unicorn/no-array-splice`

### ISSUE THREAD

**Issue by crystalfp** [sindresorhus/eslint-plugin-unicorn#3296]
The rule suggest to use `toSpliced()` instead of `splice()` even when I need to modify an array in place.

In the code below changing `structures.splice(idx, 1);` into `structures = structures.toSpliced(idx, 1);` seems to me an unneeded complication.

I'm using  "eslint-plugin-unicorn": "^67.0.0"

unicorn/no-array-splice

```js
const structure = structures[idx];
if(structure.atoms.length === 0) {
	structures.splice(idx, 1);
}
```
